src/utils/common-helpers.ts

```diff
@@ -7,3 +8,14 @@ export async function getAppRootFromConfig(server: Server) {
     return Promise.resolve('');
   }
 }
+
+export const mProjectRoot = memoize(getProjectParentRoot);
+
+/**
+ * Find the top level root of the project.
+ */
+export function getProjectParentRoot(root: string, appRoot: string) {
```

Comment:
cold we use projectForUri(uri).root for this needs? or it's some different cases?

Reply:
https://github.com/lifeart/ember-language-server/blob/component-context-info-origin/src/project-roots.ts#L100 root definition